tests/test_util.py

```diff
@@ -190,10 +190,10 @@ def test_includes_single_variable_name_replacing_slashes(self):
         )
 
     def test_decodes_encoded_chars(self):
-        url = 'https://example.com/fake-path/GPM_3IMERGHH.06%3A3B-HHR.MS.MRG.3IMERG.20200101-S120000-E122959.0720.V06B.HDF5'
+        url = 'https://example.com/fake-path/GPM_3IMERGHH.06%5D3B-HHR.MS.MRG.3IMERG.20200101-S120000-E122959.0720.V06B.HDF5'
         self.assertEqual(
             util.generate_output_filename(url),
-            'GPM_3IMERGHH.06:3B-HHR.MS.MRG.3IMERG.20200101-S120000-E122959.0720.V06B.HDF5'
+            'GPM_3IMERGHH.06]3B-HHR.MS.MRG.3IMERG.20200101-S120000-E122959.0720.V06B.HDF5'
```

Comment:
In theory we should probably replace all weird characters (non alphanumeric, dash, underscore, or dot) with underscore.  But what you have gets the job done now.